bw2h5.py

```diff
@@ -126,65 +130,74 @@ def parse_args():
 batch_ends = batch_starts + args.batch_size
 batch_ends[-1] = len(intervals)
 
-# Write batches to hdf5 file
-_logger.info('Extracting data for each batch and writing to h5 file')
+# Get output hdf5 filename
 df = None
 filename = args.prefix + '.h5'
+
+# Write batches to hdf5 file
+_logger.info('Extracting data for each batch and writing to h5 file')
 with h5py.File(filename, 'w') as f:
     # Create a single dataset -- expand locations as we go along
     for i in range(batches_per_epoch):
+
+        # Print current batch
         if i % 10 == 0:
-            _logger.info("batch: " + str(i) + " of " + str(batches_per_epoch))
+            _logger.info("batch " + str(i) + " of " + str(batches_per_epoch))
+
+        # Create dictionary to store data
+        batch_data = {}
 
         # Subset intervals
         batch_intervals = intervals.iloc[batch_starts[i]:batch_ends[i], :]
 
         # Read noisy data
-        batch_data = extract_bigwig_intervals(
+        batch_data['input'] = extract_bigwig_intervals(
             batch_intervals, args.noisybw, pad=args.pad)
 
+        # Add other input layers
+        if args.layersbw is not None:
+            # Read additional layers
+            layer_files = gather_files_from_cmdline(args.layersbw, ".bw")
+            for layer_file in layer_files:
```

Comment:
Yes, it should be the same order in which the files are passed in. Or, if a path to a directory is given, it should be the order of files in the directory.

Reply:
Added help message in `bw2h5.py`

Reply:
Order of file in the directory is a bit ambiguous though. I think it's a good idea to enforce strict ordering from the user when passing the layers.

Reply:
How would you do that? Sounds like this may need to be a separate PR?

Reply:
One way we can do it is name each of the layers individually when creating them. Then when training the model specify the ordering of the input based on the names of the layers. e.g. when creating the dataset we do something like `--layers low_cov:path_to_file_a quality_score:path_to_file_b ...`, and this will create separate keys in the h5 file called `low_cov` and `quality_score`. Now during training we can specify in model params config which of these layers you want to train with. This way we can generate a big h5 with all layers ahead of time, and train multiple models with different layers easily (for ablation studies for instance).

I would say we should do it in this PR since the API for the current implementation will be different than the one I'm suggesting. I'll let @ntadimeti take a call on this

Reply:
I agree, we should do it in this PR. Great idea @tijyojwad

Reply:
@tijyojwad added naming of layers.

Reply:
Awesome, changes look really good! Thanks @avantikalal